recipe/build-mpi.sh

```diff
@@ -75,27 +82,31 @@ fi
             --with-mpi-moduledir='${includedir}' \
             --with-wrapper-ldflags="${wrapper_ldflags}" \
             --with-sge \
+            --with-pmix=internal \
             --with-hwloc=$PREFIX \
             --with-libevent=$PREFIX \
             --with-zlib=$PREFIX \
-            --enable-mca-dso \
             --enable-ipv6 \
             $build_with_ucx \
+            $build_with_ucc \
             $build_with_cuda \
     || (cat config.log; false)
 
-make -j"${CPU_COUNT:-1}"
+# Building with 16 CPUs
+make -j16
```

Comment:
I don't think we should hardcode like this. These packages are built on CI, which doesn't have 16 cpus (at most 2, I think). Over subscribing like this often slows things down and greatly increases memory usage, sometimes to the point of crashing on low-resource CI. It should already be building with all available cpus by default, via CPU_COUNT. Is this not what you see?

Reply:
Hi Min,
It was originally building with `make -j"${CPU_COUNT:-1}"` and that took ages, but we can obviously put that back

Reply:
Were you building outside conda build? Conda build sets CPU_COUNT to the total number of cpus. If you are running this script outside, you can set CPU_COUNT=16, for example to build with 16 threads.

Reply:
Hi,

I was building on a VM, but that's Ok now: I set it back how it was, and I can see that on the Azure DevOps it builds in fact much faster than on my VM

Anyway, the issues I have now are for ppc64le (no ucc available) and osx, I am trying various things at the moment (although the syntax is very likely wrong)

Reply:
The build fails for linux_ppc64le: *RuntimeError: Could not get compatibility information for ucc package.  Is it one of your host dependencies?*

I tried `- ucc  # [linux and not ppc64le and not osx]` but that did not work: any idea what would be the correct syntax?

Reply:
I think I got it working now